mitosheet/mitosheet/step_performers/graph_steps/graph.py

```diff
@@ -102,8 +105,21 @@ def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Opti
         y_axis_column_ids = graph_creation["y_axis_column_ids"] if graph_creation["y_axis_column_ids"] is not None else []
         y_axis_column_headers = prev_state.column_ids.get_column_headers_by_ids(sheet_index, y_axis_column_ids)
 
-        # Validate optional parameters 
-        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() else None
+        # Validate optional parameters that are available for all graph types
+        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() and graph_creation["color"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
```

Comment:
I think a helper function called `get_column_header_from_optional_column_id_graph_param` might be really nice here. It is just like:
```
def get_column_header_from_optional_column_id_graph_param(state: State, params: Dict[str, Any], param_name: string) -> Optional[ColumnHeader]:
    # TODO
```